engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterLaunchEngine.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERLAUNCHENGINE_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERLAUNCHENGINE_H_
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngine.h"
+
+/**
+ * A lazy container for an engine that will only dispense one engine.
+ *
+ * This is used to hold an engine for plugin registration when the
+ * GeneratedPluginRegistrant is called on a FlutterAppDelegate before the first
+ * FlutterViewController is set up. This is the typical flow after the
+ * UISceneDelegate migration.
+ */
+@interface FlutterLaunchEngine : NSObject
+
+/**
+ * Accessor for the launch engine.
```

Comment:
Maybe also explain that this will lazily create the engine if not already there?

Reply:
> My thought is that that is an implementation detail that doesn't need to be communicated here. 

the "laziness" is part of the contract tho (i.e. because of this, the client can get the engine without extra prior call to create the engine). 

Also, it's documented in `FlutterLaunchEngine`'s comment: https://github.com/flutter/flutter/pull/168396/files/20d396c332c5721ec02a9a5879463d9602ad5b39#diff-42fec7fa0a00a2987b5edd1b3d228a888bed81cd4d4c84fdc55d6ca6b7d22046R10-R11

Reply:
Unresolving in case you missed it

Reply:
Sorry I did miss that.  I think I'm resolving these on stale github renders?

Done, added a noncommittal notice to the docstring.